src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -206,6 +213,137 @@ def get_role_policy(self, role_name, policy_name: str | None = None, attached_po
                     policy_actions.append(AWSPolicyAction("s3", privilege, f"s3a://{match.group(1)}"))
         return policy_actions
 
+    def add_uc_role(self, role_name, policy_name, s3_prefixes: set[str], account_id: str, kms_key=None):
+        aws_role_trust_doc: dict = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": "arn:aws:iam::414351767826:role/unity-catalog-prod-UCMasterRole-14S5ZJVKOTYTL"
+                    },
+                    "Action": "sts:AssumeRole",
+                    "Condition": {"StringEquals": {"sts:ExternalId": "0000"}},
+                }
+            ],
+        }
+        aws_policy_kms: dict = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Action": [
+                        "s3:GetObject",
+                        "s3:PutObject",
+                        "s3:DeleteObject",
+                        "s3:ListBucket",
+                        "s3:GetBucketLocation",
+                    ],
+                    "Resource": "",
+                    "Effect": "Allow",
+                },
+                {
+                    "Action": ["kms:Decrypt", "kms:Encrypt", "kms:GenerateDataKey*"],
+                    "Resource": ["arn:aws:kms:<KMS-KEY>"],
+                    "Effect": "Allow",
+                },
+                {
+                    "Action": ["sts:AssumeRole"],
+                    "Resource": ["arn:aws:iam::<AWS-ACCOUNT-ID>:role/<AWS-IAM-ROLE-NAME>"],
```

Comment:
Bug: account id and role names must be variables

Reply:
Fixed it

Reply:
Left over from old code. Fixed